### PR TITLE
[JBIDE-19146] load legacy connections

### PR DESCRIPTION
**Pull request by adietish**

### CODE REVIEW

**Comment by alexeykazakov** [comment]
Looks good to me.

File: plugins/org.jboss.tools.openshift.express.core/src/org/jboss/tools/openshift/express/internal/core/connection/ExpressConnectionPersistency.java [hunk 1]
```diff
@@ -25,7 +29,10 @@
 
 	@Override
 	protected String[] loadPersisted() {
-		return ExpressCorePreferences.INSTANCE.loadConnections();
+		List<String> connections = new ArrayList<String>();
+		connections.addAll(Arrays.asList(ExpressCorePreferences.INSTANCE.loadConnections()));
+		connections.addAll(Arrays.asList(ExpressCorePreferences.INSTANCE.loadLegacyConnections()));
+		return connections.toArray(new String[connections.size()]);
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.express.core/src/org/jboss/tools/openshift/express/internal/core/connection/ExpressConnectionPersistency.java]: that's a convoluted way to concatenate arrays. Rather use commons-lang ArrayUtils.addAll(first, second);

Comment (adietish) [inline plugins/org.jboss.tools.openshift.express.core/src/org/jboss/tools/openshift/express/internal/core/connection/ExpressConnectionPersistency.java]: ok, good catch, thx

**Comment by fbricon** [comment]
+1 to apply once arrays concatenation is fixed

**Comment by maxandersen** [comment]
why is there such thing as legacy connections ? didn't we keep the old format even for the new ones ?

**Comment by adietish** [comment]
This was where in some old version connections were stored. The current impl was not reading thos, something I now added.

**Comment by maxandersen** [comment]
so its fixing and additional issue ?

**Pull request merged**